pkg/controllers/provisioning/provisioner.go

```diff
@@ -212,53 +213,48 @@ var ErrNodePoolsNotFound = errors.New("no nodepools found")
 
 //nolint:gocyclo
 func (p *Provisioner) NewScheduler(ctx context.Context, pods []*corev1.Pod, stateNodes []*state.StateNode) (*scheduler.Scheduler, error) {
-	nodePoolList := &v1.NodePoolList{}
-	err := p.kubeClient.List(ctx, nodePoolList)
+	nodePools, err := nodepoolutils.List(ctx, p.kubeClient, nodepoolutils.WithManagedFilter(p.cloudProvider))
 	if err != nil {
-		return nil, fmt.Errorf("listing node pools, %w", err)
+		return nil, fmt.Errorf("listing nodepools, %w", err)
 	}
-	nodePoolList.Items = lo.Filter(nodePoolList.Items, func(n v1.NodePool, _ int) bool {
-		if !n.StatusConditions().IsTrue(status.ConditionReady) {
-			log.FromContext(ctx).WithValues("NodePool", klog.KRef("", n.Name)).Error(err, "nodePool not ready")
+	nodePools = lo.Filter(nodePools, func(np *v1.NodePool, _ int) bool {
+		if !np.StatusConditions().IsTrue(status.ConditionReady) {
+			log.FromContext(ctx).WithValues("NodePool", klog.KRef("", np.Name)).Error(err, "ignoring nodepool, not ready")
 			return false
 		}
-		return n.DeletionTimestamp.IsZero()
+		return np.DeletionTimestamp.IsZero()
 	})
-	if len(nodePoolList.Items) == 0 {
+	if len(nodePools) == 0 {
 		return nil, ErrNodePoolsNotFound
 	}
 
 	// nodeTemplates generated from NodePools are ordered by weight
 	// since they are stored within a slice and scheduling
 	// will always attempt to schedule on the first nodeTemplate
-	nodePoolList.OrderByWeight()
+	nodepoolutils.OrderByWeight(nodePools)
 
 	instanceTypes := map[string][]*cloudprovider.InstanceType{}
 	domains := map[string]sets.Set[string]{}
-	var notReadyNodePools []string
-	for _, nodePool := range nodePoolList.Items {
-		// Get instance type options
-		instanceTypeOptions, err := p.cloudProvider.GetInstanceTypes(ctx, lo.ToPtr(nodePool))
+	for _, np := range nodePools {
```

Comment:
It looks like a lot of renaming that we could skip to reduce diff.

Reply:
+1, I'd agree -- not blocking but it does make it harder to review